### PR TITLE
#15946 the reindex needs some time to be executed on move folder, it …

### PR DESCRIPTION
**Pull request by jdotcms**
…is added as a flushed cache

### CODE REVIEW

File: dotCMS/src/main/java/com/dotmarketing/portlets/browser/ajax/BrowserAjax.java [hunk 1]
```diff
@@ -740,6 +740,58 @@ public boolean copyFolder ( String inode, String newFolder ) throws Exception {
...
+			return Optional.ofNullable(errorString);
+		}
+
+		if (!folderAPI.move(currentFolder, parentFolder, user, respectFrontendRoles)) { //A folder with the same name already exists on the destination
+
+			return Optional.ofNullable(errorString);
+		}
+
+		this.addRefreshIndexCommitListener(parentFolder,null, currentFolder );
+		APILocator.getPermissionAPI().resetPermissionReferences(currentFolder);
```
Comment (wezell) [inline dotCMS/src/main/java/com/dotmarketing/portlets/browser/ajax/BrowserAjax.java]: It seems like all this should be under the folderAPI - including user permission check, cache flush and permission reset

**Review by wezell — CHANGES_REQUESTED** [review]

File: dotCMS/src/main/java/com/dotmarketing/portlets/folders/business/FolderAPI.java [hunk 2]
```diff
@@ -429,8 +429,38 @@ Folder createFolders(String path, Host host, User user, boolean respectFrontEndP
...
+	 * Move the currentFolder to non-existing folder.
+	 * @param newFolder     {@link String} destination
+	 * @param currentFolder {@link Folder} folder to move to the destination
+	 * @param user			{@link User}
+	 * @param respectFrontendRoles {@link Boolean}
+	 * @return boolean true if successfully removed
+	 * @throws DotDataException
+	 * @throws DotSecurityException
+	 */
+	boolean moveWhenDestinationDoesNotExists(final String newFolder,
```
Comment (wezell) [inline dotCMS/src/main/java/com/dotmarketing/portlets/folders/business/FolderAPI.java]: Can we make these methods private in the folder API and just have the move method call them as needed?  I feel like we are breaking encapsulation forcing decisions on API users that they should not have to make.

Also, this private method should be called:
`moveWhenDestinationDoesNotExist` singular or just 'moveToNewDestination`

**Review by wezell — CHANGES_REQUESTED** [review]
Can we make the `moveWhenDestinationDoesNotExists moveWhenDestinationDoesNotExists

File: dotCMS/src/main/java/com/dotmarketing/portlets/folders/business/FolderAPI.java (hunk 2 above)
Comment (jdotcms) [reply dotCMS/src/main/java/com/dotmarketing/portlets/folders/business/FolderAPI.java]: ok

**Review by jgambarios — APPROVED** [review]

**Review by jgambarios — APPROVED** [review]

**Pull request merged**